plugins/regex_remap/regex_remap.cc

```diff
@@ -363,19 +372,19 @@ RemapRegex::initialize(const std::string &reg, const std::string &sub, const std
           break;
         default:
           TSError("[%s] configuration variable '%s' is of an unsupported type", PLUGIN_NAME, opt_name.c_str());
-          delete cur;
           return false;
         }
         TSDebug(PLUGIN_NAME, "Overridable config %s=%s", opt_name.c_str(), opt_val.c_str());
         cur->key  = key;
         cur->type = type;
         cur->next = nullptr;
+        auto tmp  = cur.get();
```

Comment:
If I change it to this:
`        if (nullptr == last_override) {
          _first_override = cur.release();
        } else {
          last_override->next = cur.release();
        }
        last_override = cur.get();
`
last_override will be set to null (because release() clears the unique_ptr).  I could make it:
`        if (nullptr == last_override) {
          _first_override = cur.get();
        } else {
          last_override->next = cur.get();
        }
        last_override = cur.release();
`
that would work.  But, if someone added an error check like this:
`        if (nullptr == last_override) {
          _first_override = cur.get();
        } else {
          last_override->next = cur.get();
        }
        if (somethingBad) {
          return false;
        }
        last_override = cur.release();
`
then the Override instance in the heap would get doubly deleted.